Mathlib/Topology/Algebra/Valued/ValuativeRel.lean

```diff
@@ -88,6 +109,90 @@ theorem hasBasis_nhds_zero :
 @[deprecated (since := "2025-08-01")]
 alias _root_.ValuativeTopology.hasBasis_nhds_zero := hasBasis_nhds_zero
 
+variable (R) in
+lemma hasBasis_nhds_zero_ne_zero :
+    (𝓝 (0 : R)).HasBasis (· ≠ 0) fun γ ↦ { x | (valuation R) x < γ } :=
+  (hasBasis_nhds_zero R).to_hasBasis
+    (fun γ _ ↦ ⟨γ, γ.ne_zero, subset_refl _⟩)
+    fun γ hγ ↦ ⟨Units.mk0 γ hγ, trivial, subset_refl _⟩
+
+variable (R) in
+lemma hasBasis_nhds_zero_ne_zero_and_le_one :
+    (𝓝 (0 : R)).HasBasis (fun γ ↦ γ ≠ 0 ∧ γ ≤ 1) fun γ ↦ { x | (valuation R) x < γ } :=
+  (hasBasis_nhds_zero_ne_zero R).to_hasBasis
+    (fun γ hγ ↦ ⟨min γ 1, ⟨zero_lt_iff.1 <| lt_min (zero_lt_iff.2 hγ) zero_lt_one,
+      min_le_right _ _⟩, fun x hx ↦ hx.trans_le (a := v x) (min_le_left _ _)⟩)
+    fun γ hγ ↦ ⟨γ, hγ.1, subset_refl _⟩
+
+variable (R) in
+lemma hasBasis_nhds_zero_pair :
+    (𝓝 (0 : R)).HasBasis (fun rs : R × R ↦ rs.1 ∈ posSubmonoid R ∧ rs.2 ∈ posSubmonoid R)
```

Comment:
Do we prefer `r ∈ posSubmonoid R` over `¬ r ≤ᵥ 0` or `0 <ᵥ r`?

Reply:
I think we should definitely provide the lemmas `0 <ᵥ 1` and `0 <ᵥ a -> 0 <ᵥ b -> 0 <ᵥ a * b`.